pygis/docs/e_interpolation.md

```diff
@@ -0,0 +1,603 @@
+```
+
+Finally, we import the raster, extract the interpolated values of the raster to our random points, mask it to the counties boundaries, and plot the data.
+
+```{code-cell} ipython3
+# Open raster
+raster_sk = rasterio.open("../_static/e_vector_shapefiles/outputs/bay-area-rain_sk_kriging.tif")
+
+# Create copy of random points GeoDataFrame
+random_points_sk_krig = random_points.copy()
+
+# Extract raster value at each random point and add the values to the GeoDataFrame
+random_points_sk_krig["VALUE_Predict"] = [x[0] for x in raster_sk.sample(random_coords_list)]
+
+# Display attribute table
+print("Attribute Table: Random Points Interpolated Values - Scikit-Learn Kriging Method")
+display(random_points_sk_krig)
+```
+
+```{code-cell} ipython3
+# Mask raster to counties shape
+out_image_sk, out_transform_sk = rasterio.mask.mask(raster_sk, counties.geometry.values, crop = True)
+
+# Stylize plots
+plt.style.use('bmh')
+
+# Plot data
+fig, ax = plt.subplots(1, figsize = (10, 10))
+show(out_image_sk, ax = ax, transform = out_transform_sk, cmap = "RdPu")
+ax.plot(x_rain, y_rain, 'k.', markersize = 2, alpha = 0.5)
+random_points_sk_krig.plot(ax = ax, color = 'dimgray', markersize = 100)
+counties.plot(ax = ax, color = 'none', edgecolor = 'dimgray')
+plt.gca().invert_yaxis()
+
+# Set title
+ax.set_title('San Francisco Bay Area - Interpolating Rainfall using Kriging from Scikit-Learn', fontdict = {'fontsize': '15', 'fontweight' : '3'})
+
+# Display plot
+plt.show()
+```
+
+### Method 2 - Using `PyKrige`
+
+The [`pykrige` module](https://geostat-framework.readthedocs.io/projects/pykrige/en/stable/index.html) offers ordinary and universal kriging. It also supports various variogram models in addition to Gaussian.
+
+```{code-cell} ipython3
+# Adapted from: https://geostat-framework.readthedocs.io/projects/pykrige/en/latest/examples/04_krige_geometric.html
+
+# Create a 100 by 100 grid
+# Horizontal and vertical cell counts should be the same
+XX_pk_krig = np.linspace(min_x_rain, max_x_rain, 100)
+YY_pk_krig = np.linspace(min_y_rain, max_y_rain, 100)
+
+# Generate ordinary kriging object
+OK = OrdinaryKriging(
+    np.array(x_rain),
+    np.array(y_rain),
+    value_rain,
+    variogram_model = "linear",
+    verbose = False,
+    enable_plotting = False,
+    coordinates_type = "geographic",
+)
+
+# Evaluate the method on grid
+Z_pk_krig, sigma_squared_p_krig = OK.execute("grid", XX_pk_krig, YY_pk_krig)
+
+# Export raster
+export_kde_raster(Z = Z_pk_krig, XX = XX_pk_krig, YY = YY_pk_krig,
+                  min_x = min_x_rain, max_x = max_x_rain, min_y = min_y_rain, max_y = max_y_rain,
+                  proj = proj, filename = "../_static/e_vector_shapefiles/outputs/bay-area-rain_pk_kriging.tif")
```

Comment:
create a folder temp in the main pygis folder. I added it to the .gitignore so anything there won't be pushed to github. _static is for permanent files we want backed up